draft-lenders-core-coap-dtls-svcb.md

```diff
@@ -102,7 +102,103 @@ same manner as for any other service secured with transport layer security, as
 described in {{-svcb}}. Other authentication mechanisms are currently out of scope.
 
 # Constructing a CoAP URI from a SVCB Resource Record
-TBD Christian
+
+A CoAP client that obtains an SVCB record during the discovery of some service can construct a CoAP request message to interact with.
+Through the rules of {{Section 6.5 of RFC7252}} and {{Section 5.10.2 of RFC7252}}
+(or the equivalent sections for other transports as indicated by the ALPN, eg. {{Section 8.7 of ?RFC8323}}),
+those options can be composed into a URI,
+but contain more information than that.
+
+Requests are initially attempted according to the priorities of the records
+following {{Section 2.4.1 of RFC9460}}.
+Once a request is successful, subsequent requests follow that resolution path
+until the service can not be reached,
+at which point the complete discovery may be reinitiated,
+respecing any rate limits and cachable information along the way.
+
+## Input data
+
+As its input, this algorithm takes a SVCB DNS resource record set (RRs),
+consisting of information about
+
+a. the host name that was queried
+b. the service name that was queried,
+c. whether a / which non-default service port was queried,
+- a set of SVCB records, each containing
+  c. a priority,
+  d. a target name,
+  e. service parameters, and
+  f. additional information about the target (e.g. in the form of AAAA records)
+
+In a DNS request, items a, b and c are grouped using the Attrleaf naming pattern as described in {{Section 2.3 of RFC9460}}.
+
+The algorithm may also be started from equivalent data.
+For example, discovery of a network-designated resolver (DNR, {{-dnr}}) through DHCPv6
+implies that the service name that was queried was "dns" (with no queried port number),
+that the target name was the queried host name ("."),
+and contains a service priority,
+an authentication-domain-name (used as the host name, and by the previous implication, the target),
+IPv6 addresses (equivalent to AAAA records for the target)
+and service parameters.
+
+Records are required to contain information about the protocol through which they are used.
+Currently, the only way this information can be transported is through the ALPN inside the service parameters.
+Future specifications may describe other per-record parameters,
+or use service names for which a default transport is specified.
+Records with no usable transport (lack of indication of unknown ALPN values) MUST be ignored.
+
+## Request construction
+
+The steps through which the CoAP request is constructed are as follows:
+
+* Select a CoAP transport based on per-record information, falling back to a default for the named service if one is defined.
+
+  In particular, select CoAP over UDP if the ALPN is "co", and CoAP over TCP if the ALPN is "coap".
+  Note that the use of ALPNs in the presence of a default transport for the service is also influenced by the no-default-alpn parameter as per {{Section 7.1 of RFC9460}}.
```

Comment:
What if there is no ALPN? E.g. a coap transport is specified through [yet to be determined mechanism] but it is neither `coaps://...` nor `coaps+tcp://`.